### PR TITLE
fixing disabling tests bug (#1126)

### PR DESCRIPTION
**Pull request by the-moisrex**
Fixing #1126.

### CODE REVIEW

**Comment by jfalcou** [comment]
I think this is exactly what is needed.

**Review by jfalcou — APPROVED** [review]

File: CMakeLists.txt [hunk 1]
```diff
@@ -36,13 +36,13 @@ endif()
...
-include(config/compiler)
-include(config/types)
-include(config/pch)
-
 if( EVE_BUILD_TEST )
+  include(CTest)
+  include(target/add_parent_target)
+  include(config/compiler)
+  include(config/types)
+  include(config/pch)
```
Comment (the-moisrex) [inline CMakeLists.txt:44]: This file which is included here adds "documentation" related executable while the if above will make sure that executable won't be created.

Meaning that I think we're breaking things for someone who disables "tests" and enables "documentation" here.

Comment (jfalcou) [reply CMakeLists.txt:44]: As said in the issue, this documentation source thing is gonna be removed by the doc PR anyway so only test and bench options will remain.

Comment (jfalcou) [reply CMakeLists.txt:44]: See https://github.com/jfalcou/eve/pull/1124/files?file-filters%5B%5D=.cmake&hide-deleted-files=true

Comment (the-moisrex) [reply CMakeLists.txt:44]: Oh, that's okay then. thanks

**Pull request merged**